tests/sparsity/dynamic_sparsity_pattern_23.cc

```diff
@@ -0,0 +1,68 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2021 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+
+// test that iterating over a DynamicSparsityPattern with begin()/end() also
+// works for indices exceeding 32 bit integers
+
+#include <deal.II/lac/dynamic_sparsity_pattern.h>
+
+#include "../tests.h"
+
+
+void
+test()
+{
+  // take a bit more indices than we would have for 32 bit integers
+  IndexSet set((1ULL << 32) + 4);
```

Comment:
That's actually about twice as many as we have for 32-bit integers :-)

Reply:
You mean I should have said *unsigned integers*?

Reply:
The largest unsigned int is `1 << 31`, not `1 << 32`. So you actually have 8B+4 here. Not that that matters, it was just a comment on the "a bit more".